cmd/nerdctl/container_run_mount.go

```diff
@@ -228,7 +227,7 @@ func generateMountOpts(ctx context.Context, cmd *cobra.Command, client *containe
 			ociMounts[i] = x.Mount
 			mounted[filepath.Clean(x.Mount.Destination)] = struct{}{}
 
-			target, err := securejoin.SecureJoin(tempDir, x.Mount.Destination)
+			target, err := joinHostPath(tempDir, x.Mount.Destination)
```

Comment:
This issue is similar to: https://github.com/containerd/containerd/pull/8362

For non ```C:\``` volumes, we need to skip trying to copy existing contents. For ```C:``` volumes, ```containerd``` already strips the drive letter here:

https://github.com/containerd/containerd/blob/474a257b16f0aa79421038736126080696605e9d/pkg/cri/opts/container.go#L100-L101

I think it should be enough to mimic that.

Reply:
Note: a folder on the host for a potential non ```C:``` volume should already exist. We just need to skip the attempt to populate it.

Reply:
> For non C:\ volumes, we need to skip trying to copy existing contents. For C: volumes, containerd strips the drive letter

Correct, but it only trims "C:/" specifically, and AFAICT [Docker should allow for any drive other than C too](https://docs.docker.com/engine/reference/builder/#volume).

Considering the names of these host dirs seems to be irrelevant, I suggest we just use `anonVolName`. (which is actually a random ID generated on the spot)

I'll also log the mapping for some debug-ability...

Reply:
I was thinking of something like this. This should mimic what is/will (be) happening in containerd.

```suggestion
			if len(x.Mount.Destination) >= 2 && string(x.Mount.Destination[1]) == ":" && !strings.EqualFold(string(x.Mount.Destination[0]), "c") {
				continue
			}
			destination := x.Mount.Destination
			if len(destination) >= 2 && strings.EqualFold((destination[:2]), "C:") {
				destination = destination[2:]
			}
			target, err := joinHostPath(tempDir, destination)
```